Lib/test/test_concurrent_futures.py

```diff
@@ -964,6 +970,37 @@ def test_ressources_gced_in_workers(self):
         mgr.shutdown()
         mgr.join()
 
+    def test_saturation(self):
+        executor = self.executor_type(4)
+        def acquire_lock(lock):
+            lock.aquire()
+
+        mp_context = get_context()
+        sem = mp_context.Semaphore(0)
+        job_count = 15 * executor._max_workers
+        for _ in range(job_count):
+            executor.submit(acquire_lock, sem)
+        self.assertEqual(len(executor._processes), executor._max_workers)
+        for _ in range(job_count):
+            sem.release()
+        executor.shutdown()
+
+    def test_idle_process_reuse_one(self):
+        executor = self.executor_type()
+        executor.submit(mul, 21, 2).result()
+        executor.submit(mul, 6, 7).result()
+        executor.submit(mul, 3, 14).result()
+        self.assertEqual(len(executor._processes), 1)
+        executor.shutdown()
+
+    def test_idle_process_reuse_multiple(self):
+        executor = self.executor_type()
+        executor.submit(mul, 12, 7).result()
+        executor.submit(mul, 33, 25)
+        executor.submit(mul, 25, 26).result()
+        executor.submit(mul, 18, 29)
+        self.assertEqual(len(executor._processes), 2)
```

Comment:
This test might be subject to race conditions, as indicated in https://buildbot.python.org/all/#/builders/296/builds/46. Instead, I think it would probably make more sense to check `self.assertTrue(len(executor._processes) < executor._max_workers)` and set max workers to a specific amount, e.g. `executor = self.executor_type(4)`, instead of testing for an explicit number of workers. I'll take a look at this again tomorrow, after the buildbot tests have completed.